remote/remote_test.go

```diff
@@ -1496,3 +1510,61 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 }
+
+func TestRemoteRetryLogic(t *testing.T) {
+	testCases := []RemoteTestCase{
+		{name: "Do not retry after status code 200 ", repo: "org/do-not-retry", statusCode: http.StatusOK, failedCount: 0, expected: 1},
+		{name: "Retry after status code 404", repo: "org/retry-not-found", statusCode: http.StatusNotFound, failedCount: 2, expected: 3},
+		{name: "Retry after status code 401", repo: "org/retry-unauthorized", statusCode: http.StatusUnauthorized, failedCount: 2, expected: 3},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			img, _ := random.Image(1024, 1)
+			configName, _ := img.ConfigName()
+			server := createServer(img, configName, tc)
+			u, err := url.Parse(server.URL)
+			if err != nil {
+				t.Fatalf("url.Parse(%v) = %v", server.URL, err)
+			}
+			defer server.Close()
+			var repoName = u.Hostname() + ":" + u.Port() + "/" + tc.repo
+
+			// exercise the subject
+			count = 0
+			_, err = remote.NewImage(repoName, authn.DefaultKeychain, remote.WithPreviousImage(repoName))
+			if err != nil {
+				t.Fatalf("Error: %v, running remoteNewImage", err)
+			}
+			if tc.expected != count {
+				t.Fatalf("Expected number of invocations %d different than actual value %d", tc.expected, count)
+			}
+		})
+	}
+}
+
+func createServer(img v1.Image, configName v1.Hash, tc RemoteTestCase) *httptest.Server {
+	manifestPath := fmt.Sprintf("/v2/%s/manifests/latest", tc.repo)
+	configPath := fmt.Sprintf("/v2/%s/blobs/%s", tc.repo, configName)
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var err error
+		switch r.URL.Path {
+		case "/v2/":
+			w.WriteHeader(http.StatusOK)
+		case configPath:
+			c, _ := img.RawConfigFile()
+			_, err = w.Write(c)
```

Comment:
These seem not to be necessary for the tests to pass.

Reply:
(both `"/v2/"` and `configPath`)

Reply:
"/v2" is required, because it's the first call made by the client, but other one, the configPath I think it's not, let me run the tests without it and I can removes it if it is not necessary, I think I added because the first attempt was made using the remote.fromBaseImage method and that one was actually using the configPath, but I had an issue with the OS in the random image not set up, and I change to test it with remote.WithPreviosImage

Reply:
@natalieparellano you were right, they are not needed for the test to pass, I just removed those lines